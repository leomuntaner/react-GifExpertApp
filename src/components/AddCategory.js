import React, { useState } from 'react';
import PropTypes from 'prop-types';
export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
       // console.log(handleSubmit);
    };

    const handleSubmit = (e) => {
        e.preventDefault();

        if( inputValue.trim().length > 2 ){
            setCategories( cats => [inputValue, ...cats] );
            setInputValue('');  
        }
    };

    
    return (
        <form onSubmit={ handleSubmit } >
            <p> { inputValue } </p>
            <input type='text' onChange={ handleInputChange } value={ inputValue } />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories : PropTypes.func.isRequired
}
