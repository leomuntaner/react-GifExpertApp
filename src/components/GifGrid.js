import React from 'react';
import PropTypes from 'prop-types';
// import { getGifs } from '../helpers/getGifs';
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
    // CUSTOM HOOK
    const { data:images, loading } = useFetchGifs( category );

    return (

        <>
        <h3 className='animate__animated animate__fadeIn' > {category} </h3>

        { loading && <p className='animate__animated animate_flash'>Loading..</p> }

        <div className='card-grid'>
            {
                images.map( img => (
                    <GifGridItem 
                        key={img.id}
                        {...img} />
                    ))
                }                
            </div> 
        </>        
    );
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}