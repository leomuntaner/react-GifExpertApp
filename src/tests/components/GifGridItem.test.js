
import React from 'react';
import {GifGridItem} from '../../components/GifGridItem';
import {shallow} from 'enzyme';



describe('Test de Gif Grid Item', () => {

    const title = 'Soy el titulo';
    const url = 'http://lapaginadecarlos.com';
    const wrapper = shallow( <GifGridItem title={ title } url={ url } />);

    test('Debe mostrar Counter App correctamente', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe tener title en p ', () => {
        
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title )  

    });

    test('Debe tener title y url ', () => {
        
        const img = wrapper.find('img');
        expect( img.prop('alt')).toBe( title );
        expect( img.prop('src')).toBe( url );

    });
    
    test('Div ebe tener animate__fadeIn class ', () => {
        const clase = 'animate__fadeIn';
        const div = wrapper.find('div');
        expect( div.prop('className').includes(clase)).toBe( true );

    });








});
