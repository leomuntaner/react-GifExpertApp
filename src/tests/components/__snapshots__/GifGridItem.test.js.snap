// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de Gif Grid Item Debe mostrar Counter App correctamente 1`] = `ShallowWrapper {}`;
